psycopg/cursor_type.c

```diff
@@ -1579,6 +1583,219 @@ psyco_curs_copy_expert(cursorObject *self, PyObject *args, PyObject *kwargs)
     return res;
 }
 
+#define psyco_curs_start_replication_expert_doc \
+"start_replication_expert(command, writer=None, keepalive_interval=10) -- Start and consume replication stream with direct command."
+
+static PyObject *
+psyco_curs_start_replication_expert(cursorObject *self, PyObject *args, PyObject *kwargs)
+{
+    PyObject *res = NULL;
+    char *command;
+    static char *kwlist[] = {"command", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "s", kwlist, &command)) {
+        return NULL;
+    }
+
+    EXC_IF_CURS_CLOSED(self);
+    EXC_IF_GREEN(start_replication_expert);
+    EXC_IF_TPC_PREPARED(self->conn, start_replication_expert);
+
+    if (self->repl_started) {
+        psyco_set_error(ProgrammingError, self, "replication already in progress");
+        return NULL;
+    }
+
+    Dprintf("psyco_curs_start_replication_expert: command = %s", command);
+
+    self->copysize = 0;
+    self->repl_stop = 0;
+
+    self->repl_write_lsn = InvalidXLogRecPtr;
+    self->repl_flush_lsn = InvalidXLogRecPtr;
+    self->repl_apply_lsn = InvalidXLogRecPtr;
+    self->repl_feedback_pending = 0;
+
+    gettimeofday(&self->repl_last_io, NULL);
+
+    if (pq_execute(self, command, self->conn->async,
+                   1 /* no_result */, 1 /* no_begin */) >= 0) {
+        res = Py_None;
+        Py_INCREF(res);
+    }
+
+    self->repl_started = 1;
```

Comment:
I think this should be 

```
} else {
    self->repl_started = 1;
}
```

so that `repl_started` is not set if `pq_execute` returns an error.